UI connections

RoseMatt319 opened the pull request:
Finished every connected UI piece in terms of actual functionality.

RoseMatt319 committed:
Small fixes and threshold connection

Fixed gui argument being "GUI," now "--gui."
Moved all controller functionality to MainUI class from MainWindowUI.
Connected rank threshold texts to main body.

RoseMatt319 committed:
Added all connection functionality

Added connection to save and blank save boxes.
Added connection to new game box.
Added functionality for letter buttons.
Added folders for separate save types.

RoseMatt319 committed:
Fixed loading issues

Fixed found words text being broken upon loading.
Fixed letters being broken upon loading.

kingchewie reviewed (dismissed):
Everything looks in order. Really good work with the connections, your random is almost exactly the same as mine so i think it looks fine for a merge

RoseMatt319 committed:
Wrong input label and shuffle fix

Added a label for wrong inputs.
Fixed shuffle to avoid crash when no puzzle present.

RoseMatt319 committed:
Fixed new game issue

Fixed issue where starting a new game while one is in progress causes the game to have more letters than it should.

RoseMatt319 committed:
Added elements to the GUI structure

Added way to see points, separated found words from "Found Words:", made scroll bar show always.

RoseMatt319 committed:
Added text changes

Added changes to point count, points gained, and successful saves. Made text revert to blank under condition of loading a save.

RoseMatt319 committed:
Made things visually appealing

Made windows visually appealing with icons, backgrounds, and more.
Added backspace button, CC licensing window, and their functionalities.

RoseMatt319 committed:
Fixed scrolling box for found words

Box for found words now scrolls and sorts and is navigable by tab.

RoseMatt319 committed:
File and menu clean up

Cleaned up old save files.
Swapped save menu button order.

kingchewie reviewed (approved):
Appropriate post demo adjustments and cleaning/niceing up the GUI. Several connections as well. Everything looks good as best I can tell. Will put through for merge
